lib/membrane_h264_plugin/parser/au_splitter.ex

```diff
@@ -151,76 +151,59 @@ defmodule Membrane.H264.Parser.AUSplitter do
-        # and not doing that has made the application crash
-
-        nalu.type == :idr ->
-          true
-
-        nalu.parsed_fields.frame_num != last_nalu.parsed_fields.frame_num ->
-          true
-
-        nalu.parsed_fields.pic_parameter_set_id != last_nalu.parsed_fields.pic_parameter_set_id ->
-          true
-
-        Map.has_key?(nalu.parsed_fields, :field_pic_flag) and
-          Map.has_key?(last_nalu.parsed_fields, :field_pic_flag) and
-            nalu.parsed_fields.field_pic_flag != last_nalu.parsed_fields.field_pic_flag ->
-          true
-
-        Map.has_key?(nalu.parsed_fields, :field_pic_flag) and
-          Map.has_key?(last_nalu.parsed_fields, :field_pic_flag) and
-          nalu.parsed_fields.field_pic_flag == 1 and
-          last_nalu.parsed_fields.field_pic_flag == 1 and
-            nalu.parsed_fields.bottom_field_flag != last_nalu.parsed_fields.bottom_field_flag ->
-          true
-
-        nalu.parsed_fields.nal_ref_idc != last_nalu.parsed_fields.nal_ref_idc and
-            Enum.any?(
-              [nalu.parsed_fields.nal_ref_idc, last_nalu.parsed_fields.nal_ref_idc],
-              &(&1 == 0)
-            ) ->
-          true
-
-        nalu.parsed_fields.pic_order_cnt_type == 0 and
-          last_nalu.parsed_fields.pic_order_cnt_type == 0 and
-            (nalu.parsed_fields.pic_order_cnt_lsb != last_nalu.parsed_fields.pic_order_cnt_lsb or
-               Map.get(nalu.parsed_fields, :delta_pic_order_cnt_bottom) !=
-                 Map.get(last_nalu.parsed_fields, :delta_pic_order_cnt_bottom)) ->
-          true
-
-        nalu.parsed_fields.pic_order_cnt_type == 1 and
-          last_nalu.parsed_fields.pic_order_cnt_type == 1 and
-            (Bunch.Access.get_in(nalu.parsed_fields, [:delta_pic_order_cnt, 0]) !=
-               Bunch.Access.get_in(last_nalu.parsed_fields, [:delta_pic_order_cnt, 0]) or
-               Bunch.Access.get_in(nalu.parsed_fields, [:delta_pic_order_cnt, 1]) !=
-                 Bunch.Access.get_in(last_nalu.parsed_fields, [:delta_pic_order_cnt, 1])) ->
-          true
-
-        # The following condition to be checked is also specified in the documentation: "IdrPicFlag is equal to 1 for
-        # both and idr_pic_id differs in value".
-        # At the same time, it seems that it describes the situation, that we are having two different IDR frames - and
-        # that should imply, that their frame_num should differ, which is already checked in the second condition
-
-        true ->
-          false
-      end
-    else
-      false
-    end
+  defguardp frame_num_differs(a, b) when a.frame_num != b.frame_num
+
+  defguardp pic_parameter_set_id_differs(a, b)
+            when a.pic_parameter_set_id != b.pic_parameter_set_id
+
+  defguardp field_pic_flag_differs(a, b) when a.field_pic_flag != b.field_pic_flag
+
+  defguardp bottom_field_flag_differs(a, b) when a.bottom_field_flag != b.bottom_field_flag
```

Comment:
Here it's getting even more complicated :/
For this condition, the specification says that: 
"bottom_field_flag is present in both and differs in value"

But for the `bottom_field_flag` it also says that:
"When this syntax element is not present for the current slice, it shall be inferred to be equal to 0"

I assume, that even though there is a default value for `bottom_field_flag` to be inferred, it shouldn't be used in the condition, and that makes the fields' defaults list inconsistent, as we would need to just define `defaults: [field_pid_flag: 0]`, despite both `field_pid_flag` and `bottom_field_flag` have their defaults.